### PR TITLE
Incrementor support - Parses

### PR DESCRIPTION
**Pull request by wavid-b**
Incrementor added to AST and passes all test cases thus far.

### CODE REVIEW

File: src/parser/types.rs
Comment (wavid-b) [inline src/parser/types.rs:1]: can probably get rid of the object from the parser::types.rs and from stuff referencing Statement::Incrementor

File: src/parser/whamm.pest [hunk 1]
```diff
@@ -63,8 +63,8 @@ fn_call = { ID ~ "(" ~ ( arg )? ~ ( "," ~ arg )* ~ ")" }
 // var ops
 declaration = { TYPE ~ ID }
 assignment = { ID ~ "=" ~ (ternary | expr) }
-statement = { ( ( declaration | assignment | fn_call ) ~ ";" )+ }
-
+statement = { ( ( declaration | assignment | fn_call | incrementor ) ~ ";" )+ }
```
Comment (ahuoguo) [inline src/parser/whamm.pest:66]: I will recommend not changing the definition of statements, but change `expr` instead.  Define `++` as a postfix. See https://www.cl.cam.ac.uk/techreports/UCAM-CL-TR-924.pdf page 24.

File: src/parser/tests.rs [hunk 2]
```diff
@@ -122,6 +134,15 @@ map<i32, i32> count;
     r#"wasm:bytecode:call:alt  / i == """" / { }"#,
     // bad statement
     "wasm:bytecode:call:alt / i == 1 / { i; }",
+    // bad incrementor
+    r#"
+    wasm:bytecode:br:before {
+        i = 0;
+        if(i++ == 0){
```
Comment (ahuoguo) [inline src/parser/tests.rs:141]: I don't see why this is a bad incrementor (at least this is valid in C)

Comment (wavid-b) [reply src/parser/tests.rs:141]: It's not the wanted behavior for this DSL. Likely we want the "++" operator to not be accepted in expressions but as its own statement

File: src/parser/whamm.pest (hunk 1 above)
Comment (wavid-b) [reply src/parser/whamm.pest:66]: Original branch of work had it like that, but it becomes a nightmare in the emitter. This solution disallows some behavior, but ends up much less complex

File: src/parser/whamm_parser.rs [hunk 3]
```diff
@@ -401,6 +401,84 @@ fn stmt_from_rule(pair: Pair<Rule>, err: &mut ErrorGen) -> Result<Statement, Vec
...
+                loc: Some(Location {
+                    line_col: var_id_line_col.clone(),
+                    path: None,
+                }),
+            };
+            let _val = Value::Integer {
+                ty: DataType::I32,
+                val: 1,
+            };
+            let _rhs = Expr::Primitive {
```
Comment (ejrgilbert) [inline src/parser/whamm_parser.rs:420]: Prepending an underscore on a var name implies that the var isn't being used. I'd refactor to remove the underscores.

**Review by ejrgilbert — CHANGES_REQUESTED** [review]

**Comment by wavid-b** [comment]
changed the object naming - see [eaf9cdb](https://github.com/ejrgilbert/whamm/pull/45/commits/eaf9cdb256bbccf04338486c422b7d8694770182)

**Review by ejrgilbert — APPROVED** [review]

**Pull request merged**